control-plane-operator/controllers/hostedcontrolplane/ingress/router.go

```diff
@@ -62,6 +63,12 @@ func ReconcileRouterService(svc *corev1.Service, internal, crossZoneLoadBalancin
 		})
 	}
 
+	// Apply LoadBalancerSourceRanges for external router services to restrict CIDR access
+	// Only apply for external (non-internal) services and when not running on ARO HCP
+	if !internal && len(allowedCIDRBlocks) > 0 && !azureutil.IsAroHCP() {
```

Comment:
Should we just cover the AWS envs in this conditional?

Reply:
in ARO we use shared-ingress.